meilisearch-http/src/routes/document.rs

```diff
@@ -42,7 +45,10 @@ async fn get_document(
         .open_index(&path.index_uid)
         .ok_or(ResponseError::index_not_found(&path.index_uid))?;
 
-    let document_id = meilisearch_core::serde::compute_document_id(&path.document_id);
+    let document_id = update::compute_document_id(&path.document_id);
+    if !update::validate_document_id(&path.document_id) {
+        return Err(Error::Serializer(SerializerError::InvalidDocumentIdFormat).into())
+    }
```

Comment:
Why `compute_document_id` did not return a Result? Instead of validating the document_id afterward.

Reply:
You do it 3 times in that file.